### ISSUE TITLE
[Bug]: update Rsbuild to 0.0.22，use events lib error

### ISSUE THREAD

**Issue by ChenXplorer** [web-infra-dev/rspack#4643]
### Version

```sh
System:
    OS: macOS 13.5
    CPU: (10) arm64 Apple M1 Pro
    Memory: 314.17 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Browsers:
    Chrome: 119.0.6045.123
    Safari: 16.6
```

### Details

see https://github.com/web-infra-dev/rspack/issues/4597

add new steps
1. yarn add events
2. App.tsx 添加代码
```tsx
 import { useState } from 'react';
import React from 'react';
import { Button, Space } from 'antd';
++ import EventEmitter from 'events';
++ export const event = new EventEmitter();
++ event.setMaxListeners(5);
const HelloWorld = () => {
  const [count, setCount] = useState(0);

  return (
    <div>
      <h1>Rsbuild + React</h1>

      <div>
      <Button type="primary">Primary Button</Button>
    <Button>Default Button</Button>
    <Button type="dashed">Dashed Button</Button>
    <br />
    <Button type="text">Text Button</Button>
    <Button type="link">Link Butto1n</Button>
        <button type="button" onClick={() => setCount(count + 1)}>
          count is {count}
        </button>
        <p>
          Edit <code>App.tsx</code> to test HMR
        </p>
      </div>
    </div>
  );
};

export default HelloWorld;
 ```
3. error  ----->  Uncaught TypeError: Cannot read properties of undefined (reading '_events')
   actually，’this‘ is  undefined

### Reproduce link

NONE

### Reproduce Steps

SEE Details

**Comment by chenjiahan** [comment]
@bvanjoi Can you help to dig this issue? ❤️

**Comment by chenjiahan** [comment]
Reproduce steps:

1. `npm create rsbuild@latest`, select React
2. Add to index file:

```js
import EventEmitter from 'events';
export const event = new EventEmitter();
event.setMaxListeners(5);
```

3. pnpm run dev

**Comment by chenjiahan** [comment]
Fixed by https://github.com/web-infra-dev/rspack/pull/4639

**Comment by LingyuCoder** [comment]
Need to add test cases

**Comment by chenjiahan** [comment]
Canary version for this: 0.3.12-canary-bd2e9a8-20231114131339

**Comment by hardfist** [comment]
it should be fixed in rspack@0.3.13
